ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/graphql/GraphQLHttpsServiceTest.java

```diff
@@ -0,0 +1,545 @@
+
+    try (final Response resp = client.newCall(buildPostRequest(body)).execute()) {
+      Assertions.assertThat(resp.code()).isEqualTo(200); // Check general format of result
+      final JsonObject json = new JsonObject(resp.body().string());
+      testHelper.assertValidGraphQLResult(json);
+      final String result = json.getJsonObject("data").getString("gasPrice");
+      Assertions.assertThat(result).isEqualTo("0x10");
+    }
+  }
+
+  @Test
+  public void query_untypedPost() throws Exception {
+    final RequestBody body = RequestBody.create("{gasPrice}", null);
+    final Wei price = Wei.of(16);
+    Mockito.when(miningCoordinatorMock.getMinTransactionGasPrice()).thenReturn(price);
+
+    try (final Response resp = client.newCall(buildPostRequest(body)).execute()) {
+      Assertions.assertThat(resp.code()).isEqualTo(200); // Check general format of result
+      final JsonObject json = new JsonObject(resp.body().string());
+      testHelper.assertValidGraphQLResult(json);
+      final String result = json.getJsonObject("data").getString("gasPrice");
+      Assertions.assertThat(result).isEqualTo("0x10");
+    }
+  }
+
+  @Test
+  public void getSocketAddressWhenActive() {
+    final InetSocketAddress socketAddress = service.socketAddress();
+    Assertions.assertThat(socketAddress.getAddress().getHostAddress()).isEqualTo("127.0.0.1");
+    Assertions.assertThat(socketAddress.getPort()).isPositive();
+  }
+
+  @Test
+  public void getSocketAddressWhenStoppedIsEmpty() throws Exception {
+    final GraphQLHttpService service = createGraphQLHttpService();
+
+    final InetSocketAddress socketAddress = service.socketAddress();
+    Assertions.assertThat(socketAddress.getAddress().getHostAddress()).isEqualTo("0.0.0.0");
+    Assertions.assertThat(socketAddress.getPort()).isZero();
+    Assertions.assertThat(service.url()).isEmpty();
+  }
+
+  @Test
+  public void getSocketAddressWhenBindingToAllInterfaces() throws Exception {
+    final GraphQLConfiguration config = createGraphQLConfig();
+    config.setHost("0.0.0.0");
+    final GraphQLHttpService service = createGraphQLHttpService(config);
+    service.start().join();
+
+    try {
+      final InetSocketAddress socketAddress = service.socketAddress();
+      Assertions.assertThat(socketAddress.getAddress().getHostAddress()).isEqualTo("0.0.0.0");
+      Assertions.assertThat(socketAddress.getPort()).isPositive();
+      Assertions.assertThat(!service.url().contains("0.0.0.0")).isTrue();
+    } finally {
+      service.stop().join();
+    }
+  }
+
+  @Test
+  public void responseContainsJsonContentTypeHeader() throws Exception {
+
+    final RequestBody body = RequestBody.create("{gasPrice}", GRAPHQL);
+
+    try (final Response resp = client.newCall(buildPostRequest(body)).execute()) {
+      Assertions.assertThat(resp.header("Content-Type")).isEqualTo(JSON.toString());
+    }
+  }
+
+  @Test
+  public void ethGetUncleCountByBlockHash() throws Exception {
```

Comment:
can this test use getBlockNumber or something more basic? PoW is deprecated so ommers might go away at some point